sips/sip-030/sip-030-wallet-interface.md

```diff
@@ -0,0 +1,528 @@
+# Preamble
+
+SIP Number: `To be assigned`
+
+Title: Integration of a Modern Stacks Wallet Interface Standard
+
+Authors: [janniks](https://github.com/janniks), [kyranjamie](https://github.com/kyranjamie), [aryzing](https://github.com/aryzing), [m-aboelenein](https://github.com/m-aboelenein)
+
+Type: Standard
+
+Status: Draft
+
+Created: 10 October 2023
+
+License: BSD 2-Clause
+
+# OPEN QUESTIONS
+
+<!-- todo: remove section before merge -->
+
+- [ ] Should a global single `window.` object be used, or should provider discovery be handled complely differently?
+- [ ] Rename `stx_transferFt` to include SIP010 somewhere?
+- [ ] Should listeners have params? (e.g. txid for tx mined event)
+
+# Abstract
+
+This SIP proposes the integration of the standard into the Stacks blockchain's "Connect" and "Auth" system.
+The goal is to replace the current Connect interface, primarily used by web applications to connect with browser extensions and mobile apps, with a more straightforward protocol.
+The proposal consists mainly of standardizing JSON compatible interfaces for use with wallet interfaces.
+
+# Motivation
+
+The current Connect system, which has evolved since the Blockstack era, is primarily utilized by web applications for interfacing with wallets.
+However, many aspects of the existing "Connect" and "Auth" libraries are no longer required, leading to unnecessary complexity and brittleness in wallet connectivity.
+
+Recent attempts to standardize the interface have sparked valuable discussions but have not culminated in a ratified standard, largely due to the stable state of the existing system.
+This SIP aims to address these issues by adopting the WBIPs standards, which offer a more suitable RPC-style interface for modern web applications.
+The simplified protocol will allow integration without heavy dependencies (like Auth) and provider a more extendable interface for wallets.
+
+Additionally, this SIP is motivated by the increased traffic of Ordinal inscriptions on Bitcoin and the Stacks ecosystem growing closers to Bitcoin.
+The community has recognized the need for a more unified approach to wallet connectivity (e.g. Bitcoin and PSBTs for previously Stacks-only wallets).
+By adopting the new standard, we aim to align the community towards a common and modern protocol for wallet interaction in web applications.
```

Comment:
this is interesting - does this proposal only affect these web application interactions, or would offline users also benefit from this change?

Reply:
Theoretically, the RPC standard can be used anywhere. e.g. we could have the CLI options align with the RPC as well. But web is the largest driver (although the sip intends to be more or less runtime agnostic)